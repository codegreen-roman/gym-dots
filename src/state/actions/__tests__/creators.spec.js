import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {
    startWorkout,
    setStartedWorkout,
    startWorkoutWithCountdown
} from '../creators'

const mockStore = configureMockStore([thunk])

describe('General Creators', () => {

    describe('.startWorkout', () => {
        it('should return action with a type of WORKOUT_STATUS', () => {
            expect(startWorkout()).toMatchSnapshot()
        })
    })

    describe('.setStartedWorkout', () => {
        it('should return action with a type of WORKOUT_STATUS', () => {
            expect(setStartedWorkout()).toMatchSnapshot()
        })
    })

    describe('.startWorkoutWithCountdown', () => {

        const store = mockStore({ defaults: {} })

        it('creates 2 WORKOUT_STATUS actions when dispatching', (done) => {

            const promise = store.dispatch(startWorkoutWithCountdown())

            jest.runAllTimers()

            promise.then(() => {
                expect(store.getActions()).toMatchSnapshot()
                done()
            })
        })
    })
})
