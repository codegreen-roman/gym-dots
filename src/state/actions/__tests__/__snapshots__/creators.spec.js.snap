// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Creators .setStartedWorkout should return action with a type of WORKOUT_STATUS 1`] = `
Object {
  "payload": Object {
    "status": "started",
  },
  "type": "WORKOUT_STATUS",
}
`;

exports[`General Creators .startWorkout should return action with a type of WORKOUT_STATUS 1`] = `
Object {
  "payload": Object {
    "status": "starting",
  },
  "type": "WORKOUT_STATUS",
}
`;

exports[`General Creators .startWorkoutWithCountdown creates 2 WORKOUT_STATUS actions when dispatching 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "starting",
    },
    "type": "WORKOUT_STATUS",
  },
  Object {
    "payload": Object {
      "status": "started",
    },
    "type": "WORKOUT_STATUS",
  },
]
`;
